### ISSUE TITLE
How to make delegates for JS

### ISSUE THREAD

**Issue by MaZyGer** [paulbartrum/jurassic#168]
Hi, how to let it to learn what delegeates (or System.Action) are?

I have this code:
```
engineInterpreter.AddGlobalFunction(
    "Gameloop",
    new System.Action<Delegate>((code) => Debug.Log(code)), 
    typeof(JSScriptRunner));
```
which calls in the background with some conditions engine.SetGlobalFunction(name, value)
I get the error 

> NotImplementedException: Unsupported parameter type 'System.Delegate'.

What I want to achieve is this code in js

```
Gameloop(function()
{
  print("test");
});
```
The Debug.Log will be changed later.

**Comment by MaZyGer** [comment]
Ok I just changed delegate to object and find out it converts to "UserDefinedFunction". So I could call there the code which is ok.

**Comment by paulbartrum** [comment]
You should use `FunctionInstance` rather than `UserDefinedFunction` (`UserDefinedFunction` 
inherits from `FunctionInstance`).

**Comment by MaZyGer** [comment]
Thanks, I tried it and it works as well. But I still needed to change the whole idea.

I have multiple instances of ScriptFile (one instance per engine). The Gameloop content should be get called every frame in Unity but the problem was I couldn't get what instance actually is calling the function.

So I changed to function Gameloop() { // content }
and I use just now
```
if (jsScriptRunner.engine.HasGlobalValue("Gameloop"))
    jsScriptRunner.engine.CallGlobalFunction("Gameloop");
```
I am still testing around which way is better.
